lib/reporter.js

```diff
@@ -218,58 +218,77 @@ var CoverageReporter = function (rootConfig, helper, logger, emitter) {
-          if (coverageFailed && results) {
-            results.exitCode = 1
-          }
-        }
+    const mainDir = reporterConfig.dir || config.dir
+    const subDir = reporterConfig.subdir || config.subdir
+    const outputPath = generateOutputPath(basePath, browser.name, mainDir, subDir)
+    const remappedCoverageMap = await sourceMapStore.transformCoverage(coverageMap)
+
+    const options = helper.merge(config, reporterConfig, {
+      dir: outputPath,
+      subdir: '',
+      browser: browser,
+      emitter: emitter,
+      coverageMap: remappedCoverageMap
+    })
 
-        const context = istanbulLibReport.createContext(options)
-        const report = reports.create(reporterConfig.type || 'html', options)
+    // If config.check is defined, check coverage levels for each browser
+    if (hasOwnProperty.call(config, 'check') && !checkedCoverage[browser.id]) {
+      checkedCoverage[browser.id] = true
+      var coverageFailed = checkCoverage(browser, remappedCoverageMap)
+      if (coverageFailed && results) {
+        results.exitCode = 1
+      }
+    }
 
-        // // If reporting to console or in-memory skip directory creation
-        const toDisk = !reporterConfig.type || !reporterConfig.type.match(/^(text|text-summary|in-memory)$/)
+    const context = istanbulLibReport.createContext(options)
+    const report = reports.create(reporterConfig.type || 'html', options)
 
-        if (!toDisk && reporterConfig.file === undefined) {
-          report.execute(context)
-          return
-        }
+    // // If reporting to console or in-memory skip directory creation
+    const toDisk = !reporterConfig.type || !reporterConfig.type.match(/^(text|text-summary|in-memory)$/)
 
-        helper.mkdirIfNotExists(outputPath, function () {
-          log.debug('Writing coverage to %s', outputPath)
-          report.execute(context)
-        })
-      }))
+    if (!toDisk && reporterConfig.file === undefined) {
+      report.execute(context)
+      return
     }
+
+    helper.mkdirIfNotExists(outputPath, function () {
+      log.debug('Writing coverage to %s', outputPath)
+      report.execute(context)
+    })
+    return results
   }
 
-  this.onExit = function (done) {
+  this.onRunComplete = function (browsers) {
+    checkedCoverage = {}
+    let results = { exitCode: 0 }
+
+    const promiseCollection = reporters.map(reporterConfig =>
+      Promise.all(browsers.map(async (browser) => {
+        const res = await this.executeReport(reporterConfig, browser)
+        if (res && res.exitCode === 1) {
+          results = res
+        }
+      })))
+    promiseComplete = Promise.all(promiseCollection).then(() => results)
+    return promiseComplete
+  }
+
+  this.onExit = async function (done) {
+    const results = await promiseComplete
+    if (results && results.exitCode === 1) {
+      done(results.exitCode)
+      return
+    }
     if (typeof config._onExit === 'function') {
       config._onExit(done)
```

Comment:
Not sure where this function gets defined or where it's used, but it won't be called in case exitCode is 1.
See #420 for my approach, which still calls it.

Reply:
Are you sure that will be useful to call callback onExit? custom code can call with 1 or 0 too but you are ignore it.

Reply:
Build will be failed. Is matter to execute onExit method?

Reply:
I don't know where `_onExit` is defined or used, but with this change it's not called anymore in case the `exitCode` is `1`. That's what I wanted to note here.